src/main/java/com/minecolonies/coremod/colony/buildings/AbstractFilterableListIntrinsicCrafter.java

```diff
@@ -0,0 +1,174 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requestable.crafting.PublicCrafting;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolver;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.crafting.IRecipeStorage;
+import com.minecolonies.api.crafting.ItemStorage;
+import com.minecolonies.api.inventory.container.ContainerCrafting;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.TypeConstants;
+import com.minecolonies.coremod.colony.jobs.AbstractJobCrafter;
+import com.minecolonies.coremod.colony.requestsystem.resolvers.PrivateWorkerCraftingProductionResolver;
+import com.minecolonies.coremod.colony.requestsystem.resolvers.PrivateWorkerCraftingRequestResolver;
+import com.minecolonies.coremod.colony.requestsystem.resolvers.PublicWorkerCraftingProductionResolver;
+import com.minecolonies.coremod.colony.requestsystem.resolvers.PublicWorkerCraftingRequestResolver;
+import io.netty.buffer.Unpooled;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.inventory.container.INamedContainerProvider;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fml.network.NetworkHooks;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.BuildingConstants.CONST_DEFAULT_MAX_BUILDING_LEVEL;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_ID;
+
+/**
+ * Abstract class for all buildings which require a filterable list of allowed items.
+ */
+public abstract class AbstractFilterableListIntrinsicCrafter extends AbstractFilterableListBuilding
```

Comment:
Renamed in 7541035 to `AbstractFilterableListCrafter`, because there was really no reason to restrict this new type to only built-in recipes. Private+Public crafters that also show filterable lists are valid configurations. The lumberjack will now refuse to accept recipes himself. **HOWEVER** this means that the `AbstractFilterableListCrafter` now completely mirrors the code of `AbstractBuildingCrafter`. This is horrendeous code duplication, but I have no idea how this could be fixed without either diamond inheritance (won't do) or refactoring the whole system to use a composition-style pattern (out of the scope of this PR, honestly). If you have an idea, please let me know. Otherwise, it is what it is :-/